compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs

```diff
@@ -266,6 +266,30 @@ impl<'tcx> ConstToPat<'tcx> {
         inlined_const_as_pat
     }
 
+    fn is_ty_included_by_const_val(
+        &self,
+        const_val: ConstValue<'tcx>,
+        parent: Ty<'tcx>,
+        child: Ty<'tcx>,
+    ) -> bool {
+        if parent.eq(&child) {
```

Comment:
```suggestion
        if parent == child {
```

Are these types region-erased? Otherwise, it's somewhat risky to be using `.eq` or `==` here.

Reply:
I agree, it seems I need to utilize `﻿normalize_erasing_regions` in this context.